src/scripts/main.js

```diff
@@ -1 +1,109 @@
 'use strict';
+
+const doc = document.querySelector('body');
+
+let isFirstPromiseResolved = false;
+let isFirstPromiseRejected = false;
+
+const firstPromise = new Promise((resolve, reject) => {
+  const timeoutId = setTimeout(() => {
+    isFirstPromiseRejected = true;
+    // eslint-disable-next-line prefer-promise-reject-errors
+    reject();
+  }, 3000);
+
+  document.addEventListener(
+    'click',
+    () => {
+      if (!isFirstPromiseRejected) {
+        clearTimeout(timeoutId);
+        isFirstPromiseResolved = true;
+        resolve();
+      }
+    },
+    { once: true },
+  );
+});
+
+let secondPromiseResolved = false;
+const secondPromise = new Promise((resolve) => {
+  const handleResolve = () => {
+    if (!secondPromiseResolved) {
+      secondPromiseResolved = true;
+      resolve();
+    }
+  };
+
+  document.addEventListener('click', handleResolve, { once: true });
+
+  document.addEventListener(
+    'contextmenu',
+    (e) => {
+      e.preventDefault();
+      handleResolve();
+    },
+    { once: true },
+  );
+});
+
+let leftClick = false;
+let rightClick = false;
+
+const thirdPromise = new Promise((resolve) => {
+  document.addEventListener('click', () => {
+    leftClick = true;
+
+    if (leftClick && rightClick) {
+      resolve();
+    }
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+    rightClick = true;
+
+    if (leftClick && rightClick) {
+      resolve();
```

Comment:
The logic for resolving the third promise is correct, but it might be more efficient to use a single event listener for both 'click' and 'contextmenu' events to avoid redundancy.